msg/tools/px_generate_uorb_topic_files.py

```diff
@@ -170,40 +170,70 @@ def generate_output_from_file(format_idx, filename, outputdir, package, template
     return generate_by_template(output_file, template_file, em_globals)
 
 
-def generate_idl_file(filename_msg, outputdir, templatedir, package, includepath, ids):
+def generate_idl_file(filename_msg, msg_dir, alias, outputdir, templatedir, package, includepath, ids):
     """
     Generates an .idl from .msg file
     """
-    em_globals = get_em_globals(
-        filename_msg, package, includepath, ids, MsgScope.NONE)
-    spec_short_name = em_globals["spec"].short_name
+    # Make sure input msg directory exists:
+    if not os.path.isdir(msg_dir):
+        os.makedirs(msg_dir)
+
+    msg = os.path.join(msg_dir, filename_msg + ".msg")
+
+    if (alias != ""):
```

Comment:
This condition is not required.

Reply:
Actually it is because if the alias doesn't exist, then the spec name comes from the `em_globals` `spec`, otherwise the spec name is the alias itself.